stx-rosetta.Dockerfile

```diff
@@ -0,0 +1,147 @@
+### Build blockstack-core-sidecar API
+FROM node:13.14.0-buster as build
+
+ARG API_TAG=v0.28.0
+
+RUN apt-get -y update && apt-get -y install openjdk-11-jre-headless
+
+WORKDIR /app
+
+RUN git clone -b $API_TAG --depth 1 https://github.com/blockstack/stacks-blockchain-api.git .
+RUN echo "GIT_TAG=$(git tag --points-at HEAD)" >> .env
+RUN npm install && npm run build && npm prune --production
+# RUN npm config set unsafe-perm true && npm install --unsafe-perm
+# RUN npm run build --unsafe-perm && npm prune --production
+
+### Build stacks-node binary
+
+FROM rust:stretch as stacks-node-build
+
+ARG STACKS_TAG=v23.0.0.10-krypton
+
+RUN mkdir -p /src /stacks
+WORKDIR /src
+RUN git clone -b $STACKS_TAG --depth 1 https://github.com/blockstack/stacks-blockchain.git .
+RUN rustup target add x86_64-unknown-linux-gnu
+RUN cargo build --release --workspace=./ --target x86_64-unknown-linux-gnu
+RUN cp -R /src/target/x86_64-unknown-linux-gnu/release/. /stacks
+RUN ls /stacks
+
+### Fetch stacks-node binary
+
+### Begin building base image
+FROM ubuntu:focal
+
+SHELL ["/bin/bash", "-c"]
+
+### Install utils
+RUN apt-get update
+RUN apt-get install -y sudo curl pslist
+
+### Set noninteractive apt-get
+RUN echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections
+
+### Storage goes in /data; see https://www.rosetta-api.org/docs/standard_storage_location.html
+RUN mkdir -p /data
+
+### stacky user ###
+# see https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user
+RUN useradd -l -u 33333 -G sudo -md /data/stacky -s /bin/bash -p stacky stacky \
+    # passwordless sudo for users in the 'sudo' group
+    && sed -i.bkp -e 's/%sudo\s\+ALL=(ALL\(:ALL\)\?)\s\+ALL/%sudo ALL=NOPASSWD:ALL/g' /etc/sudoers
+ENV HOME=/data/stacky
+WORKDIR $HOME
+USER stacky
+RUN sudo chown -R stacky:stacky $HOME
+RUN mkdir /data/stacky/.bashrc.d
+
+### Node.js
+# ENV NODE_VERSION=14.13.1
+ENV NODE_VERSION=13.14.0
+RUN curl -fsSL https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash \
+    && bash -c ". .nvm/nvm.sh \
+        && nvm install $NODE_VERSION \
+        && nvm alias default $NODE_VERSION"
+ENV PATH=$PATH:/data/stacky/.nvm/versions/node/v${NODE_VERSION}/bin
+RUN node -e 'console.log("Node.js runs")'
+
+### Setup stacks-node
+COPY --from=stacks-node-build /stacks/stacks-node stacks-node/
+ENV PATH="$PATH:$HOME/stacks-node"
+
+#### Copy stacks-node mocknet config
+COPY ./stacks-blockchain/Stacks-mocknet.toml ./
```

Comment:
Should this use an `ARG` to toggle between mocknet, testnet, and (eventually) mainnet?

Reply:
Can I assume that there will be configuration files named `Stacks-{mocknet,testnet,mainnet}` in the `stacks-blockchain` directory?

This argument will also need to be passed to the `run.sh` script and the API server, probably in an environment variable or in a generated `.env` file.  At present, the string `testnet` is hard-coded in a couple of places in the rosetta code (`src/api/rosetta-constants.ts` and in the tests).

Reply:
I've modified the dockerfile so that it doesn't need a copy of the repository cloned to build, and it now copies all the `toml` files.  However, I haven't added the `ARG` yet.

Reply:
I've pushed a couple of commits to make the rosetta docker file and the rosetta code depend on the `STACKS_NETWORK` ARG (copied to the environment in the running container).  This lets users switch between mocknet, testnet, dev, and later, mainnet.